site/content/blog/carvel-vcluster.md

```diff
@@ -0,0 +1,195 @@
+not because you are unsure what Kubernetes cluster you are using.
+
+`vcluster` is a project from Loft Labs that delivers on an exciting idea: delivering the experience 
+of using an independent Kubernetes cluster within a host Kubernetes cluster's namespace. To summarize, 
+your cluster can host other clusters.
+
+The idea of vcluster is detailed in [greater detail on the vcluster webiste](https://www.vcluster.com/docs/architecture/basics), 
+but a summary from these docs is as follows for what vclusters are:
+
+> the basic idea is that vclusters run as a single pod (scheduled by a StatefulSet) that consists of 2 containers:
+> * Control Plane: This container contains API server, controller manager and a connection (or mount) of the data store.
+> * Syncer: What makes a vcluster virtual is the fact that it does not have a scheduler. Instead, it uses a so-called syncer 
+>   which copies the pods that need to be scheduled from the vcluster to the underlying host cluster.
+
+vclusters are also a [certified Kubernetes distribution](https://github.com/cncf/landscape/pull/2187), so the experience of using these clusters is pretty indistinguishable from other clusters.
+
+Hopefully the introduction to vclusters above is exciting to you. As users of Carvel tools though, you may want 
+to understand how the experience of `kapp` looks when using vclusters.
+
+Let's walk through an example of using `kapp` to deploy and also work with vclusters. I am using vcluster version `0.5.3`.
+
+To start, you'll need a Kubernetes cluster. I am using `minikube` in this example. You will also need `helm`, `ytt`, and 
+`kapp` to fully go through this walkthrough.
+
+vclusters are packaged as a helm chart. The vcluster CLI actually uses helm quite a bit for management of vclusters. No 
+worries though on whether this plays nice with Carvel tools as we can use helm with ytt and kapp pretty smoothly.
+
+To start, we'll need to craft a simple overlay to add an annotation to the vcluster StatefulSet. Since this StatefulSet 
+creates a PersistentVolumeClaim, we want to give `kapp` permission to delete it when we are done with our vcluster. 
+
+Go ahead sand save the overlay below to a file named `vcluster-statefulset-overlay.yml`:
+
+```
+#! vcluster-statefulset-overlay.yml
+#@ load("@ytt:overlay", "overlay")
+#@overlay/match by=overlay.subset({"kind":"StatefulSet", "metadata":{"name":"my-vcluster"}}), expects=1
+---
+spec:
+  volumeClaimTemplates:
+    #@overlay/match by=overlay.index(0)
+    - metadata:
+        name: data
+        #@overlay/match missing_ok=True
+        annotations:
+          kapp.k14s.io/owned-for-deletion: ""
+```
+
+Next, let's see what `kapp` creates when a vcluster is created:
+
+```
+$ helm template my-vcluster vcluster --repo https://charts.loft.sh --set serviceCIDR=10.96.0.0/12 -n host-namespace-1 | ytt -f- -f vcluster-stateful-set-verlay.yml | kapp deploy -a vcluster -f- -c
```

Comment:
lets flip this to `kapp deploy ... -f <(helm ... | ytt ...)` so that users can confirm

Reply:
Thanks! Was wondering how to get the prompt. Changed to the format above.